### ISSUE TITLE
set_payee does not have an era delay

### ISSUE THREAD

**Issue by CurlyBracketEffect** [paritytech/substrate#11188]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

It mentions in the docs that the effects of a set_payee call will be felt a the beginning of the next era

https://github.com/paritytech/substrate/blob/2287cd80830b081bd51bd8c313c6213d312e541e/frame/staking/src/pallet/mod.rs#L1112
https://paritytech.github.io/polkadot/pallet_staking/pallet/struct.Pallet.html#method.set_payee

However, in practice, the effects appear to be instant. For example, on Westend, the Stash set the Controller as the RewardDestination when calling [bond](https://westend.subscan.io/extrinsic/10181940-2). The Controller then called [set_payee](https://westend.subscan.io/extrinsic/10296615-2) and set the Stash as the RewardDestination. Shortly after that, they called [payout_stakers](https://westend.subscan.io/extrinsic/10296714-2) for era 4983, but the reward was [received](https://westend.subscan.io/extrinsic/10296714-2?event=10296714-39) in the Stash account when no additional eras had passed. 

I couldn't find anywhere in the code that actually implements a delay to when set_payee should be enacted, so I am inclined to believe this is an artifact from an early version of Staking, and the docs just need to be updated but wanted to double-check here before proposing any PRs.

### Steps to reproduce

See bug description for reproducible example.

**Comment by bkchr** [comment]
CC @kianenigma

**Comment by kianenigma** [comment]
yeah, seems like your conclusion is right and the doc is outdated. PR will be appreciated!
